lib/chef/provider/package/macports.rb

```diff
@@ -3,7 +3,7 @@ class Provider
     class Package
       class Macports < Chef::Provider::Package
 
-        provides :macports_package, os: "mac_os_x"
+        provides :macports_package, os: "darwin"
```

Comment:
My thought is more like this... I *can* make apt work on Redhat, and I *can* make yum work on Debian, etc. if someone goes out of their way to specifically say "I want to use macports_package", we shouldn't stand in their way. We should just never use macports for standard `package` resource on a host we don't expect it to work on.